backend/test/integration/test_schedule.py

```diff
@@ -510,3 +515,129 @@ def bust_cached_events(self, all_calendars = False):
         with with_db() as db:
             slot = repo.slot.get(db, slot_id)
             assert slot.appointment_id
+
+
+class TestDecideScheduleAvailabilitySlot:
+    start_date = datetime.now() - timedelta(days=4)
+    start_date = start_date.date()
+    start_time = time(9)
+    start_datetime = datetime.combine(start_date, start_time)
+    end_time = time(10)
+
+    def test_confirm(self, with_db, with_client, make_pro_subscriber, make_caldav_calendar, make_schedule,
+                     make_appointment, make_appointment_slot, make_attendee):
+        subscriber = make_pro_subscriber()
+        generated_calendar = make_caldav_calendar(subscriber.id, connected=True)
+        schedule = make_schedule(
+            calendar_id=generated_calendar.id,
+            active=True,
+            start_date=self.start_date,
+            start_time=self.start_time,
+            end_time=self.end_time,
+            end_date=None,
+            earliest_booking=1440,
+            farthest_booking=20160,
+            slot_duration=30)
+
+        signed_url = signed_url_by_subscriber(subscriber)
+
+        # Requested booking
+        attendee = make_attendee()
+        appointment = make_appointment(generated_calendar.id, status=models.AppointmentStatus.draft, slots=[])
+        slot: models.Slot = make_appointment_slot(appointment_id=appointment.id,
+                                                  attendee_id=attendee.id,
+                                                  booking_status=models.BookingStatus.requested,
+                                                  booking_tkn='abcd',
+                                                  )[0]
+
+        with with_db() as db:
+            # Bring the db slot to our db session
+            db.add(slot)
+            db.add(appointment)
+
+            slot_id = slot.id
+            appointment_id = appointment.id
+
+            slot.schedule_id = schedule.id
+            db.commit()
+
+            availability = schemas.AvailabilitySlotConfirmation(
+                slot_id=slot_id,
+                slot_token=slot.booking_tkn,
+                owner_url=signed_url,
+                confirmed=True
+            ).model_dump()
+
+        response = with_client.put(
```

Comment:
Routes have their own db instance so we can't have them overlap, otherwise weird stuff might happen.